frame/dapps-staking/src/tests.rs

```diff
@@ -1205,3 +1209,418 @@ fn new_era_forcing() {
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(claimer, contract, start_era, claim_era.clone());
+
+        assert_noop!(
+            DappsStaking::claim(Origin::signed(claimer), contract),
+            crate::pallet::pallet::Error::<TestRuntime>::AlreadyClaimedInThisEra
+        );
+    })
+}
+
+#[test]
+fn claim_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let claimer = 2;
+        const ERA_REWARD: Balance = 1000;
+        const STAKE_AMOUNT: Balance = 100;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+        const SKIP_ERA: EraIndex = 3;
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, 0);
+        let start_era = DappsStaking::current_era().unwrap_or(Zero::zero());
+        register(developer1, contract.clone());
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, 0);
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(claimer, contract, start_era, claim_era.clone());
+
+        cleared_contract_history(contract, FROM_ERA, claim_era);
+    })
+}
+
+#[test]
+fn claim_one_contract() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let staker1: mock::AccountId = 2;
+        let staker2: mock::AccountId = 3;
+        const ERA_REWARD: mock::Balance = 10000;
+        const STAKE_AMOUNT1: mock::Balance = 400;
+        const STAKE_AMOUNT2: mock::Balance = 600;
+        const INITIAL_STAKE: mock::Balance = 1000;
+        const NUM_OF_CONTRACTS: u64 = 1;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+        const SKIP_ERA: EraIndex = 2;
+
+        let free_staker1 = <mock::TestRuntime as Config>::Currency::free_balance(&staker1);
+        let free_staker2 = <mock::TestRuntime as Config>::Currency::free_balance(&staker2);
+        let free_developer = <mock::TestRuntime as Config>::Currency::free_balance(&developer1);
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, INITIAL_STAKE);
+        let start_era = DappsStaking::current_era().unwrap_or(Zero::zero());
+        register(developer1, contract.clone());
+        bond_and_stake(staker1, contract, STAKE_AMOUNT1);
+        bond_and_stake(staker2, contract, STAKE_AMOUNT2);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, INITIAL_STAKE);
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(staker1, contract, start_era, claim_era.clone());
+        cleared_contract_history(contract, FROM_ERA, claim_era);
+        let num_eras = 2; // number of rewarded eras
+
+        // calculate reward per stakers
```

Comment:
It would be interesting to see what happens when you have reward values which aren't so easily divisible, e.g. you have some remainder.

I assume that Balance handles this properly but still.